azext_edge/edge/providers/support/base.py

```diff
@@ -385,47 +304,72 @@ def process_storage_classes():
 
 def process_persistent_volume_claims(
     resource_api: EdgeResourceApi,
-    label_selector: str = None,
-    field_selector: str = None,
-    prefix_names: List[str] = None,
-):
-    from kubernetes.client.models import V1PersistentVolumeClaim, V1PersistentVolumeClaimList
-
+    child_group: str = "",
+    field_selector: Optional[str] = None,
+    label_selector: Optional[str] = None,
+    prefix_names: Optional[List[str]] = None,
+) -> List[dict]:
     v1_api = client.CoreV1Api()
 
-    processed = []
-    if not prefix_names:
-        prefix_names = []
-
-    pvcs: V1PersistentVolumeClaimList = v1_api.list_persistent_volume_claim_for_all_namespaces(
-        label_selector=label_selector, field_selector=field_selector
+    return _process_kubernetes_resources(
+        child_group=child_group,
+        resources=v1_api.list_persistent_volume_claim_for_all_namespaces(
+            label_selector=label_selector,
+            field_selector=field_selector
+        ),
+        resource_api=resource_api,
+        prefix_names=prefix_names,
+        kind="PersistentVolumeClaim",
     )
-    logger.info(f"Detected {len(pvcs.items)} persistent volumn claims.")
 
-    for pvc in pvcs.items:
-        d: V1PersistentVolumeClaim = pvc
-        d.api_version = pvcs.api_version
-        d.kind = "PersistentVolumeClaim"
-        pvc_metadata: V1ObjectMeta = d.metadata
-        pvc_namespace: str = pvc_metadata.namespace
-        pvc_name: str = pvc_metadata.name
 
-        if prefix_names:
-            matched_prefix = [pvc_name.startswith(prefix) for prefix in prefix_names]
-            if not any(matched_prefix):
-                continue
+def process_jobs(
+    resource_api: EdgeResourceApi,
+    child_group: str = "",
+    field_selector: Optional[str] = None,
+    label_selector: Optional[str] = None,
+    prefix_names: Optional[List[str]] = None,
+) -> List[dict]:
+    batch_v1_api = client.BatchV1Api()
+
+    return _process_kubernetes_resources(
+        child_group=child_group,
+        resources=batch_v1_api.list_job_for_all_namespaces(
+            label_selector=label_selector,
+            field_selector=field_selector
+        ),
+        resource_api=resource_api,
+        prefix_names=prefix_names,
+        kind="Job",
+    )
 
-        processed.append(
-            {
-                "data": generic.sanitize_for_serialization(obj=d),
-                "zinfo": f"{pvc_namespace}/{resource_api.moniker}/pvc.{pvc_name}.yaml",
-            }
-        )
 
-    return processed
+def process_cron_jobs(
+    resource_api: EdgeResourceApi,
+    child_group: str = "",
+    field_selector: Optional[str] = None,
+    label_selector: Optional[str] = None,
+    prefix_names: Optional[List[str]] = None,
+) -> List[dict]:
+    batch_v1_api = client.BatchV1Api()
+
+    return _process_kubernetes_resources(
+        child_group=child_group,
+        resources=batch_v1_api.list_cron_job_for_all_namespaces(
```

Comment:
easier to read if you do not do function call within function call (use a variable)